packages/bitcore-node/src/modules/ripple/api/csp.ts

```diff
@@ -0,0 +1,197 @@
+import { CSP } from '../../../types/namespaces/ChainStateProvider';
+import { InternalStateProvider } from '../../../providers/chain-state/internal/internal';
+import { RippleAPI } from 'ripple-lib';
+import { Readable } from 'stream';
+import { Storage } from '../../../services/storage';
+import { ChainNetwork } from '../../../types/ChainNetwork';
+import Config from '../../../config';
+import { FormattedTransactionType } from 'ripple-lib/dist/npm/transaction/types';
+import { ITransaction } from '../../../models/baseTransaction';
+import { ICoin } from '../../../models/coin';
+import { RippleWalletTransactions } from './transform';
+
+export class RippleStateProvider extends InternalStateProvider implements CSP.IChainStateService {
+  config: any;
+  static clients: { [network: string]: RippleAPI } = {};
+
+  constructor(public chain: string = 'XRP') {
+    super(chain);
+    this.config = Config.chains[this.chain];
+  }
+
+  async getClient(network: string) {
+    try {
+      if (RippleStateProvider.clients[network]) {
+        await RippleStateProvider.clients[network].getLedger();
+      }
+    } catch (e) {
+      delete RippleStateProvider.clients[network];
+    }
+    if (!RippleStateProvider.clients[network]) {
+      const networkConfig = this.config[network];
+      const provider = networkConfig.provider;
+      const host = provider.host || 'localhost';
+      const protocol = provider.protocol || 'wss';
+      const portString = provider.port;
+      const connUrl = portString ? `${protocol}://${host}:${portString}` : `${protocol}://${host}`;
+      RippleStateProvider.clients[network] = new RippleAPI({ server: connUrl });
+      await RippleStateProvider.clients[network].connect();
+    }
+    return RippleStateProvider.clients[network];
+  }
+
+  async getBalanceForAddress(params: CSP.GetBalanceForAddressParams) {
+    const client = await this.getClient(params.network);
+    const info = await client.getAccountInfo(params.address);
```

Comment:
@micahriggan RippleAPI will throw an error if you give it a valid address but the account has not been created (by sending it the reserve amount of XRP)

Reply:
Yeah, I've seen that one, I think we may use that to alert to the user that they'll need to deposit 21 or whatever.

Otherwise this should probably return 0.

Reply:
I would think that error should be client-side, based on a balance of 0 (and the node should return 0). Otherwise you'd have to handle this special error in insight, copay, etc.